### PR TITLE
grub2_efi: fix cross-compilation

### PR DESCRIPTION
**Pull request by delroth**
###### Motivation for this change

grub2_efi doesn't currently cross-compile because the grub build system is weird and tries to use the build toolchain to target the host. It needs explicit configuration for each tool used in order to use the proper toolchain. This is also the approach taken by buildroot, see link in comments.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/grub/2.0x.nix [hunk 1]
```diff
@@ -64,6 +64,7 @@ stdenv.mkDerivation rec {
     echo 'echo "Compile grub2 with { kbdcompSupport = true; } to enable support for this command."' >> util/grub-kbdcomp.in
   '';
 
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
```
Comment (Mindavi) [inline pkgs/tools/misc/grub/2.0x.nix:67]: Does just `cc` or `stdenv.cc`  not work here?

**Comment by Mindavi** [comment]
Diff LGTM otherwise, just the one question.

**Comment by samueldr** [comment]
We can go straight to master. GRUB, while it boots the OS, looking like it's at the root, is a pretty leafy package. A total of 23 attributes it affects in Nixpkgs proper. And the iso/sd images are *always* rebuilt anyway, so it's not like it changes anything there :)

**Review by samueldr — CHANGES_REQUESTED** [review]
I have a WIP fix currently building, but this is just a bit incomplete. It's not building the font files. So building the iso image does not work yet.

(There's also another fix needed for the iso image, which I'll send as a separate PR)

**Comment by samueldr** [comment]
Almost forgot to come back here with the fix.

With this, and changes in `iso-image.nix` part of WIP work not yet in a PR, I was able to successfully cross-compile a UEFI iso image.

Yes, `freetype` has to stay in both build inputs.

```diff
diff --git a/pkgs/tools/misc/grub/2.0x.nix b/pkgs/tools/misc/grub/2.0x.nix
index c674a17b82e..f2db5bef454 100644
--- a/pkgs/tools/misc/grub/2.0x.nix
+++ b/pkgs/tools/misc/grub/2.0x.nix
@@ -72,7 +72,7 @@ stdenv.mkDerivation rec {
   '';
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
-  nativeBuildInputs = [ bison flex python3 pkg-config autoconf automake gettext ];
+  nativeBuildInputs = [ bison flex python3 pkg-config autoconf automake gettext freetype ];
   buildInputs = [ ncurses libusb-compat-0_1 freetype lvm2 fuse libtool ]
     ++ optional doCheck qemu
     ++ optional zfsSupport zfs;
```

File: pkgs/tools/misc/grub/2.0x.nix [hunk 2]
```diff
@@ -64,6 +64,7 @@ stdenv.mkDerivation rec {
...
   nativeBuildInputs = [ bison flex python3 pkg-config autoconf automake gettext ];
```
Comment (samueldr) [inline pkgs/tools/misc/grub/2.0x.nix:68]: Almost forgot to come back here with the fix.

With this, and changes in `iso-image.nix` part of WIP work not yet in a PR, I was able to successfully cross-compile a UEFI iso image.

Yes, `freetype` has to stay in both build inputs.

```suggestion
  nativeBuildInputs = [ bison flex python3 pkg-config autoconf automake gettext freetype ];
```

**Review by samueldr — APPROVED** [review]

**Pull request merged**